pints/_mcmc/_hamiltonian.py

```diff
@@ -77,20 +80,47 @@ def __init__(self, x0, sigma0=None):
         self._n_frog_iterations = 20
 
         # Default integration step size for leapfrog algorithm
+        self._epsilon = 0.1
         self._step_size = None
-        self.set_leapfrog_step_size(0.2)
-
-        # Default masses
-        self._mass = np.ones(self._dimension)
+        self.set_leapfrog_step_size(np.diag(self._sigma0))
 
         # Divergence checking
-
         # Create a vector of divergent iterations
-        #self._divergent = np.asarray([], dtype='int')
+        self._divergent = np.asarray([], dtype='int')
 
         # Default threshold for Hamiltonian divergences
         # (currently set to match Stan)
-        #self._hamiltonian_threshold = 10**3
+        self._hamiltonian_threshold = 10**3
+
+    def set_epsilon(self, epsilon):
+        """
+        Sets epsilon for the leapfrog algorithm
+        """
+        epsilon = float(epsilon)
+        if epsilon <= 0:
+            raise ValueError('epsilon must be positive for leapfrog algorithm')
+        self._epsilon = epsilon
+        self._set_scaled_epsilon()
+
+    def epsilon(self):
+        """
+        Returns epsilon used in leapfrog algorithm
+        """
+        return self._epsilon
+
+    def scaled_epsilon(self):
+        """
+        Returns scaled epsilon used in leapfrog algorithm
+        """
+        return self._scaled_epsilon
```

Comment:
Please use e.g. `np.array(self._scaled_epsilon, copy=True)` to return a copy of the internal array (otherwise, people can modify the contents of the array using the returned reference)

Reply:
@MichaelClerx That's fine but we don't seem to do that elsewhere. If we are going to make this PINTS styles perhaps it's better to make this a separate issue?

Reply:
Fair enough!